File: 0927/1.cpp
#include <iostream>
using namespace std;

int main() {
    int n;
    while(true){
        cout << "Please input n (1<=n<=13): ";
        cin >> n;
        if(n>=1 && n<=13) break;
        else cout << "Input range error, please re-enter." << endl;
    }
    int c = 1;
    for(int i=n; i>=1; i--) {
        for(int j=1; j<=i; j++) {
            if(c<10){
                cout << "0" << c;
                c++;
            }
            else{
                cout << c;
                c++;
            }
        }
    cout << endl;
    }


    return 0;
}


// 题目描述
// 给出 n，请输出一个直角边长度是 n 的数字直角三角形。所有数字都是 2 位组成的，如果没有 2 位则加上前导 0。

// 输入格式
// 输入一个正整数 n。

// 输出格式
// 输出如题目要求的数字直角三角形。

File: 0927/1.py
def main():
    while True:
        n = int(input("Please input n (1<=n<=13): "))
        if 1 <= n <= 13:
            break
        else:
            print("Input range error, please re-enter.")

    c = 1
    for i in range(n, 0, -1):   # 从 n 到 1
        for j in range(1, i + 1):
            if c < 10:
                print(f"0{c}", end="")  # 小于 10 补 0
            else:
                print(c, end="")
            c += 1
        print()  # 换行


if __name__ == "__main__":
    main()
